### PR TITLE
RedisCluster stub fixes

### PR DESCRIPTION
**Pull request by michael-grunder**
I can now run RedisCluster unit tests within a PHP build tree build in debug mode without any deprecation warnings or arginfo/zpp errors.

### CODE REVIEW

File: redis.stub.php [hunk 1]
```diff
@@ -408,8 +408,7 @@ public function script(string $command, mixed ...$args): mixed;
 
     public function select(int $db): Redis|bool;
 
-    /** @return bool|Redis */
-    public function set(string $key, mixed $value, mixed $opt = NULL);
+    public function set(string $key, mixed $value, ?mixed $opt = NULL): Redis|string|bool;
```
Comment (nicolas-grekas) [inline redis.stub.php]: ?mixed => mixed

File: redis_cluster.stub.php [hunk 2]
```diff
@@ -72,341 +71,344 @@ public function command(mixed ...$extra_args): mixed;
...
     public function script(string|array $key_or_address, mixed ...$args): mixed;
 
-    public function sdiff(string $key, string ...$other_keys): array;
+    public function sdiff(string $key, string ...$other_keys): RedisCluster|array|false;
 
-    public function sdiffstore(string $dst, string $key, string ...$other_keys): int;
+    public function sdiffstore(string $dst, string $key, string ...$other_keys): RedisCluster|int|false;
 
-    public function set(string $key, string $value): bool;
+    public function set(string $key, mixed $value, ?mixed $options = NULL): RedisCluster|string|bool;
```
Comment (nicolas-grekas) [inline redis_cluster.stub.php]: ?mixed => mixed

File: redis_cluster.stub.php [hunk 3]
```diff
@@ -72,341 +71,344 @@ public function command(mixed ...$extra_args): mixed;
...
+    public function zlexcount(string $key, string $min, string $max): RedisCluster|int|false;
 
-    public function zpopmax(string $key, int $value = null): bool|array;
+    public function zpopmax(string $key, int $value = null): RedisCluster|bool|array;
 
-    public function zpopmin(string $key, int $value = null): bool|array;
+    public function zpopmin(string $key, int $value = null): RedisCluster|bool|array;
 
-    public function zrange(string $key, int $start, int $end, array $options = null): bool|array;
+    public function zrange(string $key, int $start, int $end, ?mixed $options_withscores = null): RedisCluster|array|bool;
```
Comment (nicolas-grekas) [inline redis_cluster.stub.php]: ?mixed => mixed

File: redis_cluster.stub.php [hunk 4]
```diff
@@ -72,341 +71,344 @@ public function command(mixed ...$extra_args): mixed;
...
+    public function flushall(string|array $node, bool $async = false): RedisCluster|bool;
 
-    public function flushdb(string|array $node, bool $async = false): bool;
+    public function flushdb(string|array $node, bool $async = false): RedisCluster|bool;
 
-    public function geoadd(string $key, float $lng, float $lat, string $member, mixed ...$other_triples): int;
+    public function geoadd(string $key, float $lng, float $lat, string $member, mixed ...$other_triples): RedisCluster|int;
 
-    public function geodist(string $key, string $src, string $dest, ?string $unit = null): array;
+    public function geodist(string $key, string $src, string $dest, ?string $unit = null): RedisClusster|float|false;
```
Comment (nicolas-grekas) [inline redis_cluster.stub.php]: RedisClusster => RedisCluster

File: redis.stub.php (hunk 1 above)
Comment (michael-grunder) [reply redis.stub.php]: Is there a formal definition for the difference between `?mixed` and `mixed` when the argument has a default value of NULL?

**Comment by nicolas-grekas** [comment]
mixed just cannot be used with anything else, the engine enforces it.

**Comment by michael-grunder** [comment]
Ahh, I understand now.  It's that `mixed` implies NULLable.  I'll make sure there are no other occurrences.

It's weird that this doesn't throw a fatal error for me when I run our unit tests.  Perhaps `declare(strict_types = 1);` might do it?

**Comment by michael-grunder** [comment]
@yatsukhnenko Since this was almost entirely changes in PHP I figured it was safe to just merge.

**Pull request merged**